crypto/s2n_tls13_keys.c

```diff
@@ -172,14 +172,21 @@ int s2n_tls13_derive_binder_key(struct s2n_tls13_keys *keys, struct s2n_psk *psk
 /*
  * Derives early secrets
  */
-int s2n_tls13_derive_early_secrets(struct s2n_tls13_keys *keys)
+int s2n_tls13_derive_early_secrets(struct s2n_tls13_keys *keys, struct s2n_psk *psk)
 {
     notnull_check(keys);
 
-    s2n_tls13_key_blob(psk_ikm, keys->size); /* in 1-RTT, PSK is 0-filled of key length */
-
     /* Early Secret */
-    GUARD(s2n_hkdf_extract(&keys->hmac, keys->hmac_algorithm, &zero_length_blob, &psk_ikm, &keys->extract_secret));
+    if (psk == NULL) {
+        /* in 1-RTT, PSK is 0-filled of key length */
+        s2n_tls13_key_blob(psk_ikm, keys->size);
+
+        GUARD(s2n_hkdf_extract(&keys->hmac, keys->hmac_algorithm, &zero_length_blob, &psk_ikm, &keys->extract_secret));
+    } else {
+        /* Sanity check that an early secret exists */
+        S2N_ERROR_IF(psk->early_secret.size == 0, S2N_ERR_SAFETY);
```

Comment:
Nit: this is what ne_check / ENSURE_NE is for https://github.com/awslabs/s2n/blob/main/utils/s2n_safety.h#L390